packages/vega-scenegraph/src/CanvasRenderer.js

```diff
@@ -31,9 +36,16 @@ prototype.initialize = function(el, width, height, origin, scaleFactor, options)
 };
 
 prototype.resize = function(width, height, origin, scaleFactor) {
+  if (this._resized) {
+    throw new Error('cannot resize canvas twice');
+  }
+
+  this._resized = true;
   base.resize.call(this, width, height, origin, scaleFactor);
-  resize(this._canvas, this._width, this._height,
-    this._origin, this._scale, this._options && this._options.context);
+  if (this._canvas) {
```

Comment:
It looks like resize will never occur with a passed-in context, as `this._canvas` will be undefined?

Reply:
Oops, there should indeed not be an 'if' here!